lib_vehicle_model/include/lib_vehicle_model/LibVehicleModel.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+/*
+ * Copyright (C) 2018-2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <memory>
+#include <stdexcept>
+#include <ros/ros.h>
+#include "ModelAccessException.h"
+#include "VehicleState.h"
+#include "VehicleMotionModel.h"
+#include "VehicleModelControlInput.h"
+#include "ParameterServer.h"
+#include "KinematicsSolver.h"
+#include "KinematicsProperty.h"
+
+namespace lib_vehicle_model {
+
+  //
+  // Public Namespace
+  //
+
+  /**
+   * @brief Initialization function for vehicle model. Loads the vehicle model as specified by parameters 
+   * 
+   * When provided with a NodeHandle this function uses the ROSParameterServer instead of a user specified parameter server
+   * 
+   * @param nh A pointer to the node handle which will be used to initialize the ROSParameterServer
+   * 
+   * @throws std::invalid_argument If the model could not be loaded or parameters could not be read
+   * @throws ModelAccessException If this function is called more than once within the same process execution
+   * 
+   */ 
+  void init(std::shared_ptr<ros::NodeHandle> nh);
```

Comment:
A node handle is something that the user ROS node is very likely to want to use in other situations such as setting up subscribers or advertisers. So there could be a shared ownership here. 

Passing by reference ties the lifeline of the lib_vehicle_model to the lifeline of the provided node handle. Since any future use of the parameter server after the node handle is destroyed would result in an exception. 

If it is changed to pass by reference, I would want to also prevent passing of the parameter server as a shared pointer such that it should only be used in the constructor.

Reply:
I think you made a good point here. Because multiple nodes are use this function group. It has to be a shared ownership.